### PR TITLE
feat(curriculum): add hash table lab

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-hash-table-class/66f1805aac05f00a7028fbcc.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+Your `HashTable` class should have a `remove` method.
+
+```js
+const table = new HashTable();
+assert.isFunction(table.remove);
+```
+
+`hashTable.remove("fCC")` should remove the `fCC` key from the `collection` object.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/lab-hash-table-class/66f1805aac05f00a7028fbcc.md:107]:
```suggestion
`hashTable.remove("FCC")` should remove the `FCC` key from the `collection` object.
```

**Review by moT01 — CHANGES_REQUESTED** [review]
This is looking good @Dario-DC 👍 

It looks like this is just a partial copy of an existing challenge? I don't see the id of this challenge in any existing challenges - or a challenge that is the exact same. I do see [this one, which is similar](https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/create-a-hash-table) - so yea, I think we're fine to just create a new id here.

Only problem I noticed was the remove user stories/tests. I originally had this as my remove method:

```
  remove(key) {
    const hashedKey = this.hash(key);
    delete this.collection[hashedKey][key];
  }
```

So when it ran, I was left with an empty object if that was the only k/v in that hashed key - tests 12 and 13 are checking that the collection `isEmpty` - it wasn't, so they failed. We could change that user story to something like: "The `remove` method should take a key as a parameter and remove the corresponding key-value pair from the hash table. If that's the last key-value pair, it should remove the hash key as well." Not sure how to best word it. The tests could maybe be reworded and/or changed slightly as well.

Let me know if you want me to try and come up with something.

**Comment by Dario-DC** [comment]
@moT01 I wouldn't say it is a partial copy because in the original challenge the `hash` function was already there ready to use. Here it becomes part of the `HashTable` class as a method. Anyway I think we should use a different ID for this lab.

I added your suggestion to clarify how `remove` should work. Let me know if that part still needs to be improved.

**Review by zairahira — COMMENTED** [review]
Went through the project, it's looking good 👍️

**Review by moT01 — APPROVED** [review]

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**